common/src/main/java/com/wynntils/core/webapi/profiles/ServerProfile.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.webapi.profiles;
+
+import java.util.Set;
+import java.util.concurrent.TimeUnit;
+
+public class ServerProfile {
+
+    long firstSeen;
+    Set<String> players;
+
+    public ServerProfile(long firstSeem, Set<String> players) {
+        this.firstSeen = firstSeem;
+        this.players = players;
+    }
+
+    public Set<String> getPlayers() {
+        return players;
+    }
+
+    public long getFirstSeen() {
+        return firstSeen;
+    }
+
+    public String getUptime() {
+        long millis = System.currentTimeMillis() - firstSeen;
+
+        return String.format(
+                "%dh %dm",
+                TimeUnit.MILLISECONDS.toHours(millis),
+                TimeUnit.MILLISECONDS.toMinutes(millis)
+                        - TimeUnit.HOURS.toMinutes(TimeUnit.MILLISECONDS.toHours(millis)));
+    }
+
+    /**
+     * This makes the firstSeem match the user computer time instead of the server time
+     *
+     * @param serverTime the input server time
+     */
+    public void matchTime(long serverTime) {
+        firstSeen = firstSeen - (System.currentTimeMillis() - serverTime);
```

Comment:
I know this is a port from legacy, but I still don't think this is correct?
```
System.currentTimeMillis() - newFirstSeen = serverTime - firstSeen
```
is the equation that maintains the change in time.
This rearranges to give
```
newFirstSeen = (System.currentTimeMillis() - serverTime) + firstSeen
```